pkg/auth/auth_handler.go

```diff
@@ -232,6 +244,17 @@ func (a *AuthHandler) CreateAuthenticator() func(name, in string, authenticate s
 				}
 				return true, nil, common.NewInfraError(http.StatusUnauthorized, err)
 			}
+			//this code is part of the authorization process and should move to authz_handler
+			//after https://github.com/go-openapi/runtime/issues/158 is resolved
+			if payload, ok := p.(*ocm.AuthPayload); ok && !payload.IsAdmin {
```

Comment:
just a matter of taste, but to reduce cascading, maybe worth to extract some vars (like payload)